eng/tools/mcp/azure-sdk-java-mcp/src/index.ts

```diff
@@ -0,0 +1,262 @@
+            title: "Build Java SDK",
+        },
+    },
+    async (args) => {
+        logToolCall("build_java_sdk");
+        return await buildJavaSdk(args.rootDirectory, args.moduleDirectory, args.groupId, args.artifactId);
+    },
+);
+
+// Tool: get_java_sdk_changelog
+server.registerTool(
+    "get_java_sdk_changelog",
+    {
+        description:
+            "Retrieve the changelog for a service submodule whose groupId starts with `com.azure`. Requires the absolute path to the JAR file, groupId, and artifactId as input parameters.",
+        inputSchema: {
+            jarPath: z
+                .string()
+                .describe(
+                    "Absolute path to the JAR file of the Java SDK, typically located in the `target` directory of the module.",
+                ),
+            groupId: z.string().describe("The Maven groupId for the Java SDK module."),
+            artifactId: z.string().describe("The Maven artifactId for the Java SDK module."),
+        },
+        annotations: {
+            title: "Get Java SDK Changelog",
+        },
+    },
+    async (args) => {
+        logToolCall("get_java_sdk_changelog");
+        const result = await getJavaSdkChangelog(args.jarPath, args.groupId, args.artifactId);
+        return result;
+    },
+);
+
+// Tool: update_java_sdk_changelog
+server.registerTool(
+    "update_java_sdk_changelog",
+    {
+        description:
+            "Update the CHANGELOG.md file for a Java SDK service submodule whose groupId starts with `com.azure`. Requires the absolute path to the JAR file, groupId, and artifactId as input parameters.",
+        inputSchema: {
+            jarPath: z
+                .string()
+                .describe(
+                    "Absolute path to the JAR file of the Java SDK, typically located in the `target` directory of the module.",
+                ),
+            groupId: z.string().describe("The Maven groupId for the Java SDK module."),
+            artifactId: z.string().describe("The Maven artifactId for the Java SDK module."),
+        },
+        annotations: {
+            title: "Update Java SDK CHANGELOG.md",
+        },
+    },
+    async (args) => {
+        logToolCall("update_java_sdk_changelog");
+        return await updateChangelogMd(args.jarPath, args.groupId, args.artifactId);
+    },
+);
+
+// Tool: instruction_migrate_typespec
+server.registerTool(
+    "instruction_migrate_typespec",
+    {
+        description:
+            "Provides step-by-step instructions for generating a Java SDK after migrating from Swagger (OpenAPI) to TypeSpec. Use this tool to understand the migration workflow and required actions.",
+        inputSchema: {},
+        annotations: {
+            title: "TypeSpec Migration Instructions",
+        },
+    },
+    async () => {
+        logToolCall("instruction_migrate_typespec");
+        return await brownfieldMigration();
+    },
+);
+
+// Tool: sync_java_sdk
+server.registerTool(
+    "sync_java_sdk",
```

Comment:
We can give this a better name. This doesn't sync the Java SDK. This copies the service TypeSpec specification from remote to local.